src/NuGet.Clients/NuGet.PackageManagement.UI/Models/DetailControlModel.cs

```diff
@@ -562,8 +562,18 @@ public DisplayVersion SelectedVersion
                             PackageMetadata = detailedPackageMetadata;
                         }
 
-                        NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(() => SelectedVersionChangedAsync(_searchResultPackage, _selectedVersion.Version, loadCts.Token).AsTask())
-                                                               .PostOnFailure(nameof(DetailControlModel));
+                        NuGetUIThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+                        {
+                            try
+                            {
+                                await SelectedVersionChangedAsync(_searchResultPackage, _selectedVersion.Version, loadCts.Token).AsTask();
+                            }
+                            catch (OperationCanceledException) when (loadCts.IsCancellationRequested)
```

Comment:
Would the `OperationCanceledException` ever be thrown by anything else?  Just curious if we need the `when` statement.

Reply:
HttpClient throws `TaskCancelledException`, which extends `OperationCancelledException`, when there's an HTTP timeout getting a request.  If we don't handle these timeouts, then we end up with missing data in our UI, which is a bug, hence I don't want to stop reporting those faults.

I think other BCL APIs that have timeouts, like `SemaphoreSlim` with a timeout, also throw TaskCancelledException, although maybe they throw System.TimeoutException. I only know for certain that HttpClient uses TaskCancelledException.